ee/rbac/lib/rbac/grpc_servers/okta_server.ex

```diff
@@ -124,6 +128,79 @@ defmodule Rbac.GrpcServers.OktaServer do
     end)
   end
 
+  def set_up_group_mapping(req, _stream) do
+    observe("set_up_group_mapping", fn ->
+      # Validate that we received a valid UUID for org_id
+      validate_uuid!(req.org_id)
+
+      # Convert from protobuf messages to our internal format (list of maps)
+      mappings =
+        Enum.map(req.mappings, fn mapping ->
+          %{
+            idp_group_id: mapping.okta_group_id,
+            semaphore_group_id: mapping.semaphore_group_id
+          }
+        end)
+
+      case Rbac.Okta.IdpGroupMapping.create_or_update(req.org_id, mappings, req.default_role_id) do
+        {:ok, _mapping} ->
+          Logger.info("Group mappings created/updated for org #{req.org_id}")
+          %SetUpGroupMappingResponse{}
+
+        {:error, %Ecto.Changeset{} = changeset} ->
+          Logger.error(
+            "Error while setting up group mappings for org #{req.org_id}: #{inspect(changeset)}"
+          )
+
+          grpc_error!(
+            :failed_precondition,
+            "Failed to save group mappings: Invalid"
+          )
+
+        error ->
+          Logger.error("Unknown error while setting up group mappings: #{inspect(error)}")
+          grpc_error!(:unknown, "Unknown error while setting up group mappings")
+      end
+    end)
+  end
+
+  def describe_group_mapping(req, _stream) do
+    observe("list_group_mappings", fn ->
+      validate_uuid!(req.org_id)
+
+      case Rbac.Okta.IdpGroupMapping.get_for_organization(req.org_id) do
+        {:ok, idp_group_mapping} ->
+          # Convert from our internal format to protobuf messages
+          group_mapping =
+            Enum.map(idp_group_mapping.group_mapping, fn %{
+                                                           idp_group_id: okta_id,
+                                                           semaphore_group_id: semaphore_id
+                                                         } ->
+              %GroupMapping{
+                okta_group_id: okta_id,
+                semaphore_group_id: semaphore_id
+              }
+            end)
```

Comment:
Can we maybe refactore this so that we don't have this formating :D Maybe we don't need pattern matching within the function here